include/revng/Support/IRHelpers.h

```diff
@@ -943,11 +943,26 @@ inline std::vector<llvm::GlobalVariable *> extractCSVs(llvm::Instruction *Call,
   return Result;
 }
 
+inline bool CompareByName(const llvm::GlobalVariable *LHS,
+                          const llvm::GlobalVariable *RHS) {
+  revng_assert(LHS->hasName() and RHS->hasName());
+  return LHS->getName() < RHS->getName();
+};
+
+template<typename RangeOfPointersToGlobal>
```

Comment:
Can we have a (simple) concept constraint? cc/@Cvelth

Reply:
We already have `range_with_value_type` just for this.

https://github.com/revng/revng/blob/3fb9e58b0a0826b52d9ae97c0e6cbb9920f71247/include/revng/ADT/Concepts.h#L143

Reply:
Eh that doesn't always work, because it assumes `R` always have `value_type`, which is not the case if e.g. `R` is an iterator range. And that's one of the ways I'm using it in `PromoteGlobalToLocalVars.cpp` (with `llvm::make_first_range`).

I changed the concept not to use `value_type` and to use the typed obtained dereferencing `R.begin()` instead. How does that look now?